### DISCUSSION TITLE
Why does the segment  merge automatically only after build an index?

### DISCUSSION THREAD

**Discussion by lvmnn** [milvus-io/milvus#22081] [Q&A and General discussion] [answered]
Is this a feature?

**Comment by yhmo** [comment, marked as answer]
"Why does the segment merge automatically"------- do you mean compaction?

**Reply by lvmnn** [reply]
> "Why does the segment merge automatically"------- do you mean compaction?

yeah

**Reply by yhmo** [reply]
Just confirmed, this is as-designed.
In v2.0/2.1, segments can be automatically compacted without index, but we found it caused some load failure bugs.
So, in v2.2, we change the logic, segments are compacted only after building an index.

**Reply by xiaofan-luan** [reply]
2.3 will remove the limit.
We add this limit in 2.2 for some temporary fix
